internal/forest/namespacestructure.go

```diff
@@ -46,6 +53,10 @@ func (ns *Namespace) SetParent(p *Namespace) {
 	// Update new parent.
 	ns.parent = p
 	if p != nil {
+		// Add usage to new subtree.
+		if nsUsage != nil {
+			ns.UseResources(nsUsage)
+		}
```

Comment:
I tested it and indeed you are correct. I'm testing the first option for now since I feel it may be clearer to understand.

Reply:
sgtm - any chance you could add your test as an integ test as well? That seems like good functionality to not regress on :)

Reply:
I'm actually having some issues with the integration testing here in general. For example [here](https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/master/internal/forest/namespacehrq_test.go#L326-L400), the setup of the test is done using [`SetParent`](https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/master/internal/forest/namespacehrq_test.go#L550). But since `SetParent` now updates the subtree usages, I'm getting the set-up wrong and therefore the integ testing fails.

I can adjust the testing but perhaps there's a better way? What do you think? @adrianludwin

Reply:
I _think_ that `testNS.subtree` is the problem. Basically, in the original (Google-internal) version of HRQ, HRQ was external to HNC, and didn't use the forest. So we used `subtree` to initialize the usages of the subtree (a namespace and all its descendants) correctly, by hand.

Now, that's not necessary, since calling `SetParent` will correctly update all the subtree usages. So maybe try modifying `buildForest` to ignore the subtrees completely, and see if that works. If it does, you can remove the `subtree` field from wherever it appears.

Reply:
Since we are setting up the hierarchy using `SetParent`, I just set `subtree` usage to be equal to `local` usage in `buildForest`. Seems to be doing the trick

Reply:
👍